dali/imgcodec/decoders/decoder_test_helper.h

```diff
@@ -134,30 +132,75 @@ class DecoderTestBase : public ::testing::Test {
   * @brief Checks if the image and the reference are equal after converting the reference
   * with ConvertSatNorm
   */
+  template <typename RefType>
+  void AssertEqualSatNorm(const TensorView<StorageCPU, OutputType> &img,
+                          const TensorView<StorageCPU, RefType> &ref) {
+    Check(img, ref, EqualConvertSatNorm());
+  }
+
   void AssertEqualSatNorm(const TensorView<StorageCPU, OutputType> &img,
                           const Tensor<CPUBackend> &ref) {
     TYPE_SWITCH(ref.type(), type2id, RefType, NUMPY_ALLOWED_TYPES, (
       Check(img, view<const RefType>(ref), EqualConvertSatNorm());
     ), DALI_FAIL(make_string("Unsupported reference type: ", ref.type())));  // NOLINT
   }
 
+  void AssertEqualSatNorm(const Tensor<CPUBackend> &img,
+                          const Tensor<CPUBackend> &ref) {
+    AssertEqualSatNorm(view<const OutputType>(img), ref);
+  }
+
   /**
   * @brief Checks if an image is close to a reference
   *
   * The eps parameter shound be specified in the dynamic range of the image.
   */
-  void AssertClose(const Tensor<CPUBackend> &img, const Tensor<CPUBackend> &ref, float eps) {
+  template <typename RefType>
+  void AssertClose(const TensorView<StorageCPU, OutputType> &img,
+                   const TensorView<StorageCPU, RefType> &ref,
+                   float eps) {
     if (std::is_integral<OutputType>::value)
       eps /= max_value<OutputType>();
+    Check(img, ref, EqualConvertNorm(eps));
+  }
+
+  void AssertClose(const TensorView<StorageCPU, OutputType> &img,
+                   const Tensor<CPUBackend> &ref,
+                   float eps) {
+    TYPE_SWITCH(ref.type(), type2id, RefType, NUMPY_ALLOWED_TYPES, (
+      Check(img, view<const RefType>(ref), EqualConvertNorm(eps));
+    ), DALI_FAIL(make_string("Unsupported reference type: ", ref.type())));  // NOLINT
+  }
+
+  void AssertClose(const Tensor<CPUBackend> &img,
+                   const Tensor<CPUBackend> &ref,
+                   float eps) {
     TYPE_SWITCH(ref.type(), type2id, RefType, NUMPY_ALLOWED_TYPES, (
       Check(view<const OutputType>(img), view<const RefType>(ref), EqualConvertNorm(eps));
     ), DALI_FAIL(make_string("Unsupported reference type: ", ref.type())));  // NOLINT
   }
 
+
   /**
   * @brief Crops a tensor to specified roi_shape, anchored at roi_begin.
   * Does not support padding.
   */
+  template <int ndim>
+  void Crop(const TensorView<StorageCPU, OutputType, ndim> &output,
```

Comment:
Sometimes we want to `Crop` the reference, which is not necessarily of type `OutputType` (when testing things such us color conversion it's convenient to keep the reference as float). What do you think about templating this `Crop` with image type as well?
```cpp
template <typename ImageType, int ndim>
void Crop(const TensorView<StorageCPU, ImageType, ndim> &output,
...
```

Reply:
done